dask_kubernetes/operator/core.py

```diff
@@ -0,0 +1,167 @@
+import subprocess
+import json
+import re
+import time
+
+from distributed.core import rpc
+from distributed.deploy import Cluster
+
+from distributed.utils import Log, Logs
+
+from dask_kubernetes.auth import ClusterAuth
+from .daskcluster import (
+    build_cluster_spec,
+    build_worker_group_spec,
+)
+
+
+class KubeCluster2(Cluster):
+    """Launch a Dask Cluster on Kubernetes using the Operator"""
+
+    def __init__(
+        self,
+        name,
+        namespace="default",
+        image="daskdev/dask:latest",
+        replicas=3,
+        resources={},
+        env={},
+        asynchronous=False,
+        auth=ClusterAuth.DEFAULT,
+        **kwargs,
+    ):
+        self.name = name
+        self.namespace = namespace
+        self.core_api = None
+        self.custom_api = None
+        self.image = image
+        self.replicas = replicas
+        self.resources = resources
+        self.env = env
+        self.auth = auth
+
+        # TODO: Check if cluster already exists
+        super().__init__(asynchronous=asynchronous, **kwargs)
+        if not self.asynchronous:
+            self._loop_runner.start()
+            self.sync(self._start)
+
+    async def _start(self):
+        data = build_cluster_spec(
+            self.name, self.image, self.replicas, self.resources, self.env
+        )
+        with open("data.json", "w") as jfile:
+            json.dump(data, jfile)
+        cluster = subprocess.check_output(
+            [
+                "kubectl",
+                "apply",
+                "-f",
+                "data.json",
+                "-n",
+                self.namespace,
+            ],
+            encoding="utf-8",
+        )
+        time.sleep(5)
+        pf = subprocess.Popen(
```

Comment:
Can we reuse [get_external_address_for_scheduler](https://github.com/dask/dask-kubernetes/blob/2aa7212cc7fbffd25caa8cf474d950dd876821d0/dask_kubernetes/utils.py#L43) here instead?